### PR TITLE
F array

### PR DESCRIPTION
**Pull request by TheRealMorgenfrue**

### CODE REVIEW

File: Project/src/main/java/com/proj4/AST/visitors/CheckVisitors/ExpressionTypeChecker.java
Comment (MrKahr) [inline Project/src/main/java/com/proj4/AST/visitors/CheckVisitors/ExpressionTypeChecker.java:1]: Remove line 194: debugging left in code

File: Project/src/test/java/com/proj4/ArrayTest.java
Comment (MrKahr) [inline Project/src/test/java/com/proj4/ArrayTest.java:1]: Implement test testing whether arrays can contain templates.

File: Project/src/test/java/com/proj4/StateDeclTest.java
Comment (MrKahr) [inline Project/src/test/java/com/proj4/StateDeclTest.java:1]: Remove debug commenting.

**Review by KristianEmilWN — APPROVED** [review]
If tests are passing this looks good!

**Review by KristianEmilWN — APPROVED** [review]
No changes needed, but remember to mention lack of complex comparison in report testing results

**Review by MrKahr — APPROVED** [review]

**Pull request merged**